### PR TITLE
Add config for CodeLlama 70B

### PR DESCRIPTION
**Pull request by awaelchli**
The 70B CodeLlama released today: https://huggingface.co/codellama/CodeLlama-70b-hf

### CODE REVIEW

File: lit_gpt/config.py [hunk 1]
```diff
@@ -866,6 +866,26 @@ def norm_class(self) -> Type:
         intermediate_size=22016,
         rope_base=1000000,
     ),
+    # https://huggingface.co/codellama/CodeLlama-70b-hf/blob/main/config.json
+    dict(
+        name="CodeLlama-70b-hf",
+        hf_config=dict(org="codellama", name="CodeLlama-70b-hf"),
+        block_size=16384,
+        vocab_size=32016,
+        padding_multiple=64,
```
Comment (awaelchli) [inline lit_gpt/config.py:875]: Need to choose this

**Comment by awaelchli** [comment]
I don't know why the freaking tokenizer wouldn't output the same token ids, despite them representing the same text tokens.

**Comment by carmocca** [comment]
The HF tokenizer is being funny. I don't know why. Let's simply merge as is and we'll look into where the difference is coming from in the future if it gives us problems or becomes a recurring issue with other checkpoints.

File: lit_gpt/config.py [hunk 2]
```diff
@@ -978,6 +1018,26 @@ def norm_class(self) -> Type:
         intermediate_size=22016,
         rope_base=1000000,
     ),
+    # https://huggingface.co/codellama/CodeLlama-70b-Instruct-hf/blob/main/config.json
+    dict(
+        name="CodeLlama-70b-Instruct-hf",
+        hf_config=dict(org="codellama", name="CodeLlama-70b-Instruct-hf"),
+        block_size=16384,
+        vocab_size=32016,
+        padding_multiple=16,
```
Comment (awaelchli) [inline lit_gpt/config.py:1027]: @carmocca would you recommend using padded_size here as well?

Comment (carmocca) [reply lit_gpt/config.py:1027]: Doesn't matter, vocab_size+padding_multiple=padded_size

**Pull request merged**